frontend/src/services/APICache.ts

```diff
@@ -1,19 +1,71 @@
 import axios, { AxiosResponse } from 'axios';
 import { env } from '@/env';
+import axiosGet from '@/utils/networkUtils';
+import { AppStateTables } from '@/types/AppState';
+
+// Type shenanigans from: https://www.reddit.com/r/typescript/comments/nv0icn/is_it_possible_to_create_an_array_union_types_as/
+export const InitialEndpoints = [
+  'crops',
+  'cropsconversionfactors',
+  'cropsoiltestphosphorousregions',
+  'cropsoilpotassiumregions',
+  'croptypes',
+  'manures',
+  'nmineralizations',
+  'previousyearmanureapplications',
+  'regions',
+  'soiltestmethods',
+  'soiltestphosphorouskelonwaranges',
+  'soiltestphosphorousranges',
+  'soiltestphosphorousrecommendation',
+  'soiltestpotassiumkelownaranges',
+  'soiltestpotassiumranges',
+  'soiltestpotassiumrecommendation',
+  'subregions',
+  'units',
+] as const;
+
+export type InitialEndpoint = (typeof InitialEndpoints)[number];
 
 class APICache {
-  private endpointCache: { [endpoint: string]: AxiosResponse } = {};
+  private initialized: boolean;
+
+  private endpointCache: { [endpoint: string]: AxiosResponse };
+
+  public promise: Promise<null>;
+
+  private async initialize() {
+    await Promise.all(
+      InitialEndpoints.map((endpoint) =>
+        axiosGet(`${env.VITE_BACKEND_URL}/api/${endpoint}/`).then((res) => {
+          if (res.status === 200) {
+            this.endpointCache[endpoint] = res;
+          }
+          // TODO: Handle errors here
```

Comment:
Do we need to deal with this TODO or is it for later?

Reply:
Oh, this is a reminder for later. I think we should have a specific error message if the initial loading fails, because that probably indicates something is wrong with the database or the user's internet.